presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/ElasticsearchMetadataFetcher.java

```diff
@@ -0,0 +1,604 @@
+            }
+        }
+        catch (IOException ex) {
+            throw new PrestoException(ELASTICSEARCH_CONNECTION_ERROR, format("Failed to close connection"), ex);
+        }
+    }
+
+    static TransportClient createNativeTransportClient(ElasticsearchConnectorConfig config, String host, int port)
+    {
+        return createNativeTransportClient(config, host, port, Optional.empty());
+    }
+
+    static TransportClient createNativeTransportClient(ElasticsearchConnectorConfig config, String host, int port, Optional<String> clusterName)
+    {
+        TransportAddress address;
+        try {
+            address = new TransportAddress(InetAddress.getByName(host), port);
+        }
+        catch (Exception e) {
+            throw new PrestoException(ELASTICSEARCH_CONNECTION_ERROR, format("Failed to establish connection to search node: %s:%s", host, port), e);
+        }
+
+        Settings settings;
+        Settings.Builder builder = Settings.builder();
+        if (clusterName.isPresent()) {
+            builder.put("cluster.name", clusterName.get());
+        }
+        else {
+            builder.put("client.transport.ignore_cluster_name", true);
+        }
+        switch (config.getCertificateFormat()) {
+            case PEM:
+                settings = builder
+                        .put(SEARCHGUARD_SSL_TRANSPORT_PEMCERT_FILEPATH, config.getPemcertFilepath())
+                        .put(SEARCHGUARD_SSL_TRANSPORT_PEMKEY_FILEPATH, config.getPemkeyFilepath())
+                        .put(SEARCHGUARD_SSL_TRANSPORT_PEMKEY_PASSWORD, config.getPemkeyPassword())
+                        .put(SEARCHGUARD_SSL_TRANSPORT_PEMTRUSTEDCAS_FILEPATH, config.getPemtrustedcasFilepath())
+                        .put(SEARCHGUARD_SSL_TRANSPORT_ENFORCE_HOSTNAME_VERIFICATION, false)
+                        .build();
+                return new PreBuiltTransportClient(settings, SearchGuardSSLPlugin.class).addTransportAddress(address);
+            case JKS:
+                settings = builder
+                        .put(SEARCHGUARD_SSL_TRANSPORT_KEYSTORE_FILEPATH, config.getKeystoreFilepath())
+                        .put(SEARCHGUARD_SSL_TRANSPORT_TRUSTSTORE_FILEPATH, config.getTruststoreFilepath())
+                        .put(SEARCHGUARD_SSL_TRANSPORT_KEYSTORE_PASSWORD, config.getKeystorePassword())
+                        .put(SEARCHGUARD_SSL_TRANSPORT_TRUSTSTORE_PASSWORD, config.getTruststorePassword())
+                        .put(SEARCHGUARD_SSL_TRANSPORT_ENFORCE_HOSTNAME_VERIFICATION, false)
+                        .build();
+                return new PreBuiltTransportClient(settings, SearchGuardSSLPlugin.class).addTransportAddress(address);
+            default:
+                settings = builder.build();
+                return new PreBuiltTransportClient(settings).addTransportAddress(address);
+        }
+    }
+
+    /**
+     * Get the shards given <i>index</i> in cluster defined by <i>table</i>.
```

Comment:
Remove this javadoc. It doesn't add any info that's not obvious from the method signature.

Same for other functions below.